### PR TITLE
Feature/messy recorder resolution

### PR DESCRIPTION
**Pull request by alexjeffburke**
This PR converts the Recorder such that it resolves with messy instances. In order to achieve that, the disk serialisation formatter was rewritten to work atop messy objects.

Doing this is a pre-requisite for having other output formats.

### CODE REVIEW

**Comment by alexjeffburke** [comment]
I can’t quite figure out where the coverage drop is coming from but the tests all pass after the last commit :)

@papandreou I think this probably ready to get a thorough once over now - if everything is well with the formatter then there should be no outward behavioural change so this is more internally preparing for other serialisation options.

**Comment by papandreou** [comment]
> I can’t quite figure out where the coverage drop is coming from but the tests all pass after the last commit :)

Coveralls have a "new lines missed in diff" section now that sheds a light on that: https://coveralls.io/builds/24703881

Looks like it's mostly `formatRecordedExchange.js`: https://coveralls.io/builds/24703881/source?filename=lib%2FformatRecordedExchange.js#L39

File: lib/UnexpectedMitmRecorder.js [hunk 1]
```diff
@@ -137,89 +72,75 @@ class UnexpectedMitmRecorder {
...
-                      Object.keys(responseResult.headers).forEach(
-                        headerName => {
-                          res.setHeader(
-                            headerName,
-                            responseResult.headers[headerName]
-                          );
-                        }
+                  setImmediate(() => {
+                    res.statusCode = responseResult.statusCode;
+                    Object.keys(responseResult.headers).forEach(headerName => {
```
Comment (papandreou) [inline lib/UnexpectedMitmRecorder.js]: Can be converted to `for...of` while we're at it.

**Comment by alexjeffburke** [comment]
@papandreou I addressed those two review comments.

Then I felt kinda bad about the coverage, so I wrote some thorough tests of the formatter and promptly caught some issues :D

So err, what was the thing about again?

**Pull request merged**